systest/tests/partition_test.go

```diff
@@ -0,0 +1,224 @@
+			require.IsType(t, &spacemeshv1.GlobalStateData_GlobalState{}, data)
+
+			resp := data.(*spacemeshv1.GlobalStateData_GlobalState)
+			layer := resp.GlobalState.Layer.Number
+			if layer > stop {
+				return false, nil
+			}
+
+			stateHash := types.BytesToHash(resp.GlobalState.RootHash)
+			tctx.Log.Debugw("state hash collected",
+				"client", client.Name,
+				"layer", layer,
+				"state", stateHash.ShortString())
+			stateCh <- &stateUpdate{
+				layer:  layer,
+				hash:   stateHash,
+				client: client.Name,
+			}
+			return true, nil
+		})
+	}
+
+	finalErr := eg.Wait()
+	close(stateCh)
+	// this map contains info for debugging only
+	hashes := make(map[uint32]map[types.Hash32][]string)
+	// this map contains the latest hash reported by each client/layer
+	latestStates := make(map[string]map[uint32]types.Hash32)
+	for update := range stateCh {
+		if _, ok := hashes[update.layer]; !ok {
+			hashes[update.layer] = make(map[types.Hash32][]string)
+		}
+		if _, ok := hashes[update.layer][update.hash]; !ok {
+			hashes[update.layer][update.hash] = []string{}
+		}
+		hashes[update.layer][update.hash] = append(hashes[update.layer][update.hash], update.client)
+
+		if _, ok := latestStates[update.client]; !ok {
+			latestStates[update.client] = make(map[uint32]types.Hash32)
+		}
+		latestStates[update.client][update.layer] = update.hash
+	}
+	for layer := uint32(layersPerEpoch * 2); layer <= stop; layer++ {
+		tctx.Log.Debugw("client states",
+			"layer", layer,
+			"num_states", len(hashes[layer]),
+			"states", log.ObjectMarshallerFunc(func(encoder log.ObjectEncoder) error {
+				for hash, clients := range hashes[layer] {
+					encoder.AddString("hash", hash.ShortString())
+					encoder.AddInt("num_clients", len(clients))
+					encoder.AddArray("clients", log.ArrayMarshalerFunc(func(encoder log.ArrayEncoder) error {
+						for _, c := range clients {
+							encoder.AppendString(c)
+						}
+						return nil
+					}))
+				}
+				return nil
+			}))
+	}
+	refState := latestStates[cl.Client(0).Name]
+	pass := true
+	for i := 1; i < cl.Total(); i++ {
+		clientState := latestStates[cl.Client(i).Name]
+		agree := true
+		for layer := uint32(layersPerEpoch * 2); layer <= stop; layer++ {
+			if clientState[layer] != refState[layer] {
+				tctx.Log.Errorw("client state differs from ref state",
+					"client", cl.Client(i).Name,
+					"ref_client", cl.Client(0).Name,
+					"layer", layer,
+					"client_hash", clientState[layer],
+					"ref_hash", refState[layer])
+				agree = false
+			}
+		}
+		if agree {
+			tctx.Log.Debugw("client agreed with ref client on all layers",
+				"client", cl.Client(i).Name,
+				"ref_client", cl.Client(0).Name)
+		}
+		pass = pass && agree
+	}
+	if finalErr != nil {
+		tctx.Log.Errorw("test failed", "err", finalErr.Error())
+	}
+	require.NoError(t, finalErr)
+	require.True(t, pass)
+	_ = txeg.Wait()
+}
+
+func TestPartition_30_70(t *testing.T) {
+	t.Parallel()
+
+	tctx := testcontext.New(t, testcontext.Labels("sanity"))
```

Comment:
we may have to remove it from sanity group. and create a separate job for lets say destructive group. or use more resources and parallize tests better

Reply:
> create a separate job

do you mean to have two systest jobs running concurrently?

Reply:
one by one. for concurrent we need to provision more clusters at the same time